### ISSUE TITLE
Datasource snowflake_database not returning values 

### ISSUE THREAD

**Issue by WobblyRobbly** [Snowflake-Labs/terraform-provider-snowflake#1882]
**Provider Version**

v0.62.0

**Terraform Version**

1.4.4

**Description**

We are seeing an issue where the database level retention policy is being set to a default 1. The provider is querying Snowflake to get the database details correctly however it is then unable to parse those results into the required struct. Snowflake now returns an additional field, `kind`. The datasource does not return a value so the value just defaults to 1

e.g.

```
data "snowflake_database" "this" {
  name = var.database_name
}

locals {
retention_time = var.data_retention_days != null ? var.data_retention_days : data.snowflake_database.this.retention_time
}

resource "snowflake_schema" "schema" {
  database            = var.database_name
  name                = local.schema_name
  comment             = "Terraform-managed Schema"
  is_transient        = false
  is_managed          = var.is_managed
  data_retention_days = local.retention_time
}
```

Unfortunately the provider doesn't error and carries on without logging an error, although if we set
export TF_LOG=DEBUG

we can see

```
2023-06-14T21:06:38.909+0100 [INFO]  provider.terraform-provider-snowflake: 2023/06/14 21:06:38 [DEBUG] sql-conn-query [query SHOW DATABASES LIKE 'DEV_DB' err <nil> duration 106.320233ms args {}]: timestamp=2023-06-14T21:06:38.908+0100
2023-06-14T21:06:38.909+0100 [INFO]  provider.terraform-provider-snowflake: 2023/06/14 21:06:38 [DEBUG] no databases found: timestamp=2023-06-14T21:06:38.909+0100
2023-06-14T21:06:38.909+0100 [INFO]  provider.terraform-provider-snowflake: 2023/06/14 21:06:38 [DEBUG] list database failed to decode: timestamp=2023-06-14T21:06:38.909+0100
```

We can see a successful `SHOW DATABASES LIKE 'DEV_DB'` in the Query History

A couple of potential fixes

- Add kind into the various definitions and functions
- Use Unsafe() to allow any new fields to be ignore until explicitly added in future versions of the provider

This looks to have been introduced by the change in Snowflake Behaviour documented [Behaviour Change](https://community.snowflake.com/s/article/7-19-Behavior-Changes-June-5-8-2023) in 2023_03

**Comment by malduarte** [comment]
Probably outside of the scope for this MR, but I'd argue that provider code should use Unsafe() across the board (or change to another row mapping approach) so that the provider is not tightly coupled with the current snowflake release

**Comment by mlorek** [comment]
should be fixed by #1872

**Comment by WobblyRobbly** [comment]
This is working as of 0.67.0

Closing
